# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../iputils/open_max.patch
# Copyright (C) 2009 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

http://cvs.fedoraproject.org/viewvc/rpms/iputils/F-10/iputils-20070202-open-max.patch?revision=1.2&view=markup

diff -up iputils-s20071127/rdisc.c.open_max iputils-s20071127/rdisc.c
--- iputils-s20071127/rdisc.c.open_max	2008-02-25 11:15:37.000000000 +0100
+++ iputils-s20071127/rdisc.c	2008-02-25 11:17:30.000000000 +0100
@@ -240,14 +240,25 @@ void do_fork(void)
 {
 	int t;
 	pid_t pid;
+	long open_max;
 
 	if (trace)
 		return;
+	if ((open_max = sysconf(_SC_OPEN_MAX)) == -1) {
+		if (errno == 0) {
+			(void) fprintf(stderr, "OPEN_MAX is not supported\n");
+		} 
+		else {
+			(void) fprintf(stderr, "sysconf() error\n");
+		}
+		exit(1);
+	}
+
 
 	if ((pid=fork()) != 0)
 		exit(0);
 
-	for (t = 0; t < OPEN_MAX; t++)
+	for (t = 0; t < open_max; t++)
 		if (t != s)
 			close(t);
 

